### PR TITLE
Fixed network bugs in common core dev

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.h [hunk 1]
```diff
@@ -26,7 +26,4 @@
 
 @interface MSIDAADRequestErrorHandler : NSObject <MSIDHttpRequestErrorHandling>
 
-@property (nonatomic) NSInteger retryCounter;
-@property (nonatomic) NSTimeInterval retryInterval;
-
```
Comment (jasoncoolmax) [inline IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.h]: These two lines have been removed but seems added back accidentally by this PR:
https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/commit/c087868f1c23c7347f24bf4ac4d2bfa7cc67cbf5#diff-31bccab38d271abf0c141fbf8856d8de

File: IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.m [hunk 2]
```diff
@@ -47,38 +48,64 @@ - (void)handleError:(NSError * )error
...
-    [additionalInfo setValue:httpResponse.allHeaderFields forKey:MSIDHTTPHeadersKey];
-    [additionalInfo setValue:[NSString stringWithFormat: @"%ld", (long)httpResponse.statusCode]          forKey:MSIDHTTPResponseCodeKey];
+    [additionalInfo setValue:httpResponse.allHeaderFields
+                      forKey:MSIDHTTPHeadersKey];
+
+    [additionalInfo setValue:[NSString stringWithFormat: @"%ld", (long)httpResponse.statusCode]
+                      forKey:MSIDHTTPResponseCodeKey];
     
-    NSError *httpError = MSIDCreateError(MSIDHttpErrorCodeDomain, MSIDErrorServerUnhandledResponse, errorDescription, oauthError, suberror, nil, context.correlationId, additionalInfo);
+    NSError *httpError = MSIDCreateError(MSIDHttpErrorCodeDomain, MSIDErrorServerUnhandledResponse, errorDescription, nil, nil, nil, context.correlationId, additionalInfo);
```
Comment (jasoncoolmax) [inline IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.m]: Currently we don't have logic to generate errors like `MSIDErrorServerInvalidScope`, do we need to add it here? :)

File: IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.h (hunk 1 above)
Comment (antrix1989) [reply IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.h]: yeah, those lines were added by mistake

File: IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.m (hunk 2 above)
Comment (oldalton) [reply IdentityCore/src/network/error_handler/MSIDAADRequestErrorHandler.m]: As discussed offline, it's handled lines 65-89 :)

**Pull request merged**